### PR TITLE
# Fix Compatibility with NOTMODIFYOTHERENTITYFOURNPRICE const.

### PR DESCRIPTION
**Pull request by mapiolca**
# Fix Compatibility with NOTMODIFYOTHERENTITYFOURNPRICE hidden const with multicompany.

Before : if product fourn price shared, all entity can modify them.

Now : if admin wants, all entities can display product fourn price shared, but not modify them.

### CODE REVIEW

**Comment by mapiolca** [comment]
@hregis do you think it's ok do to that ?

File: htdocs/product/fournisseurs.php [hunk 1]
```diff
@@ -933,7 +934,14 @@ function update_price_from_multicurrency() {
 
 						// Modify-Remove
 						print '<td class="center nowraponall">';
-						if ($usercancreate)
+						
+						if (!empty($conf->global->NOTMODIFYOTHERENTITYFOURNPRICE) && $productfourn->product_fourn_entity!=$conf->entity) {
+							$canmodifyotherentityfournprice = 0;
+						}else{
+							$canmodifyotherentityfournprice = 1;
+						}
```
Comment (hregis) [inline htdocs/product/fournisseurs.php]: @mapiolca 
This verification of access rights must be done in the hook "doAction" of the Multicompany module.

Comment (eldy) [reply htdocs/product/fournisseurs.php]: Nothing to do. Regis means that this must be done by the module (by adding a doAction hook) so not into the code of Dolibarr. So you can set canmodifyotherentityfournprice to 1 at begin of page before the action, keep the test on the var in core and let the module decide to overwrite this variable or not (this is a policy/feature to implement in module multicompany).

Comment (mapiolca) [reply htdocs/product/fournisseurs.php]: Ok, @hregis. Does-it integer in Multicompany module ?

**Pull request merged**